spec/rubocop/cop/style/map_into_array_spec.rb

```diff
@@ -0,0 +1,345 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::MapIntoArray, :config do
+  it 'registers an offense and corrects when using `each` with `<<` to build an array' do
+    expect_offense(<<~RUBY)
+      dest = []
+      src.each { |e| dest << e * 2 }
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `map` instead of `each` to map elements into an array.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      src.map { |e| e * 2 }
```

Comment:
Shouldn’t the expected correction be
```suggestion
      dest = src.map { |e| e * 2 }
```
Removing a local variable is bound to break people’s code.

Reply:
Fixed and amended!